alicloud/resource_alicloud_maxcompute_project.go

```diff
@@ -0,0 +1,136 @@
+package alicloud
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/maxcompute"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudMaxComputeProject() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAliyunMaxComputeProjectCreate,
+		Read:   resourceAliyunMaxComputeProjectRead,
+		Delete: resourceAliyunMaxComputeProjectDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(3, 27),
+			},
+
+			"specification_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"OdpsStandard"}, false),
```

Comment:
If specification_type only support one valid value, please set it to optional and add a default for it.